src/tss2-fapi/api/Fapi_Provision.c

```diff
@@ -932,15 +932,46 @@ Fapi_Provision_Finish(FAPI_CONTEXT *context)
             if ((*capabilityData)->data.tpmProperties.tpmProperty[0].value == VENDOR_INTC) {
                 /* Get INTEL certificate for EK public hash via web */
                 uint8_t *cert_buffer = NULL;
-                size_t cert_size;
+                char *cert_start = NULL, *cert_end = NULL, *cert_bin = NULL;
+                size_t cert_size = 0, buffer_size;
                 TPM2B_PUBLIC public;
                 r = ifapi_get_intl_ek_certificate(context, &pkey->public, &cert_buffer,
-                                                  &cert_size);
+                                                  &buffer_size);
                 goto_if_error_reset_state(r, "Get certificates", error_cleanup);
 
-                r = ifapi_cert_to_pem(cert_buffer, cert_size, &command->pem_cert,
-                                      NULL, &public);
+                LOGBLOB_DEBUG(cert_buffer, buffer_size, "%s", "Certificate");
+                /* Parse certificate data out of XML */
+                cert_start = strstr((char *)cert_buffer, "\"certificate\":\"");
```

Comment:
Since this is very intl-specific, this json and b64 decoding should be done inside the get_intl_cert.c

Reply:
moved to get_intl_cert.c